code/modules/mob/living/carbon/human/species_types/golems.dm

```diff
@@ -990,12 +990,45 @@
 			H.reagents.remove_reagent(chem.type, chem.volume - 5)
 			to_chat(H, "<span class='warning'>The excess milk is dripping off your bones!</span>")
 		H.heal_bodypart_damage(1.5,0, 0)
-		H.reagents.remove_reagent(chem.type, REAGENTS_METABOLISM)
+		H.reagents.remove_reagent(chem.type, chem.metabolization_rate)
 		return TRUE
-
 	if(chem.type == /datum/reagent/toxin/bonehurtingjuice)
+		H.adjustStaminaLoss(7.5, 0)
 		H.adjustBruteLoss(0.5, 0)
-		H.reagents.remove_reagent(chem.type, REAGENTS_METABOLISM)
+		if(prob(20))
+			switch(rand(1, 3))
+				if(1)
+					var/list/possible_says = list("oof.", "ouch!", "my bones.", "oof ouch.", "oof ouch my bones.")
+					H.say(pick(possible_says), forced = /datum/reagent/toxin/bonehurtingjuice)
+				if(2)
+					var/list/possible_mes = list("oofs softly.", "looks like their bones hurt.", "grimaces, as though their bones hurt.")
+					H.say("*custom " + pick(possible_mes), forced = /datum/reagent/toxin/bonehurtingjuice)
+				if(3)
+					to_chat(H, "<span class='warning'>Your bones hurt!</span>")
+		if(chem.volume >= chem.overdose_threshold)
+			chem.overdosed = TRUE

```

Comment:
I was thinking putting this where the check_reagent proc is called, not have to manually do it per each reagent.

For handle_chemicals all you should have to do is check to see if `chem.overdosed` is T